lib/Crypto/Encryption.php

```diff
@@ -189,7 +189,23 @@ public function begin($path, $user, $mode, array $header, array $accessList) {
 		// always use the version from the original file, also part files
 		// need to have a correct version number if they get moved over to the
 		// final location
-		$this->version = (int)$this->keyManager->getVersion($this->stripPartFileExtension($path), new View());
+		if (\OC::$server->getSession()->exists('renameFile')) {
```

Comment:
while this might have worked for confirming the fix, we need a clean approach to detect renames.

Storing in the session is an absolute no go. Imagine if there are concurrent file operations where one is a rename, the value stored in the session would affect any other parallel unrelated file transfers and would likely destroy the known universe